homeassistant/components/ituran/coordinator.py

```diff
@@ -0,0 +1,77 @@
+"""Coordinator for Ituran."""
+
+import logging
+
+from pyituran import Ituran, Vehicle
+from pyituran.exceptions import IturanApiError, IturanAuthError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONF_ID_OR_PASSPORT,
+    CONF_MOBILE_ID,
+    CONF_PHONE_NUMBER,
+    DOMAIN,
+    UPDATE_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+type IturanConfigEntry = ConfigEntry[IturanDataUpdateCoordinator]
+
+
+class IturanDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Vehicle]]):
+    """Class to manage fetching Ituran data."""
+
+    config_entry: IturanConfigEntry
+    ituran: Ituran
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize account-wide Ituran data updater."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{DOMAIN}-{entry.data[CONF_ID_OR_PASSPORT]}",
+            update_interval=UPDATE_INTERVAL,
+            config_entry=entry,
+        )
+        self.ituran = Ituran(
+            entry.data[CONF_ID_OR_PASSPORT],
+            entry.data[CONF_PHONE_NUMBER],
+            entry.data[CONF_MOBILE_ID],
+        )
+
+    async def _async_update_data(self) -> dict[str, Vehicle]:
+        """Fetch data from Ituran."""
+
+        try:
+            vehicles = await self.ituran.get_vehicles()
+        except IturanApiError as e:
+            raise UpdateFailed(
+                e, translation_domain=DOMAIN, translation_key="api_error"
+            ) from e
+        except IturanAuthError as e:
+            raise ConfigEntryAuthFailed(
+                e, translation_domain=DOMAIN, translation_key="auth_error"
```

Comment:
```suggestion
                translation_domain=DOMAIN, translation_key="api_error"
            ) from e
        except IturanAuthError as e:
            raise ConfigEntryAuthFailed(
                translation_domain=DOMAIN, translation_key="auth_error"
```
either pass as translation placeholder or not at all